pkg/s3select/select.go

```diff
@@ -302,9 +302,11 @@ func (s3Select *S3Select) Open(getReader func(offset, length int64) (io.ReadClos
 		s3Select.recordReader, err = csv.NewReader(s3Select.progressReader, &s3Select.Input.CSVArgs)
 		if err != nil {
 			rc.Close()
+			if strings.Contains(err.Error(), "bzip2 data invalid: bad magic value") {
```

Comment:
you should use, this covers all bzip2 structural errors.

```go
var stErr bzip2.StructuralError
if errors.As(err, &stErr) {
    return errInvalidBZIP2CompressionFormat(err)
}
```

Reply:
OK